src/sentry/api/endpoints/organization_details.py

```diff
@@ -954,7 +967,40 @@ def put(self, request: Request, organization) -> Response:
             with transaction.atomic(router.db_for_write(Organization)):
                 organization, changed_data = serializer.save()
 
-            if "targetSampleRate" in changed_data:
+            sampling_mode = organization.get_option("sentry:sampling_mode", SAMPLING_MODE_DEFAULT)
+            is_org_mode = sampling_mode == DynamicSamplingMode.ORGANIZATION.value
+            is_project_mode = sampling_mode == DynamicSamplingMode.PROJECT.value
+
+            if is_org_mode and "targetSampleRate" in changed_data:
+                organization.update_option(
+                    "sentry:target_sample_rate", serializer.validated_data["targetSampleRate"]
+                )
+                boost_low_volume_projects_of_org_with_query.delay(organization.id)
+
+            if "samplingMode" in changed_data and request.access.has_scope("org:write"):
+                with transaction.atomic(router.db_for_write(ProjectOption)):
+                    if is_project_mode:
+                        for project in organization.project_set.all():
+                            current_rate, _ = get_boost_low_volume_projects_sample_rate(
+                                org_id=organization.id,
+                                project_id=project.id,
+                                error_sample_rate_fallback=None,
+                            )
+                            if current_rate:
+                                project.update_option("sentry:sampling_rate", current_rate)
```

Comment:
Same for the other places where this option is modified.

```suggestion
                                project.update_option("sentry:target_sampling_rate", current_rate)
```